library/std/src/sync/rwlock/tests.rs

```diff
@@ -501,3 +501,108 @@ fn panic_while_mapping_write_unlocked_poison() {
 
     drop(lock);
 }
+
+#[test]
+fn test_downgrade_basic() {
+    let r = RwLock::new(());
+
+    let write_guard = r.write().unwrap();
+    let _read_guard = RwLockWriteGuard::downgrade(write_guard);
+}
+
+#[test]
+fn test_downgrade_observe() {
+    // Taken from the test `test_rwlock_downgrade` from:
+    // https://github.com/Amanieu/parking_lot/blob/master/src/rwlock.rs
+
+    const W: usize = if cfg!(target_pointer_width = "64") { 100 } else { 20 };
```

Comment:
This is still too many threads, running the test in miri takes *minutes*. Please consider that most CPUs only have ~8 cores, so that's the maximum of threads that will be run in parallel anyway.
```suggestion
    const W: usize = 20;
```